src/sentry/static/sentry/app/components/accordion.jsx

```diff
@@ -0,0 +1,83 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import styled from 'react-emotion';
+import InlineSvg from 'app/components/inlineSvg';
+import space from 'app/styles/space';
+import {Flex} from 'grid-emotion';
+import Badge from 'app/components/badge';
+import Button from 'app/components/buttons/button';
+
+class AccordionButton extends React.Component {
+  static propTypes = {
+    labelClosed: PropTypes.string.isRequired,
+    labelOpen: PropTypes.string,
+    cutoff: PropTypes.number,
+    children: PropTypes.arrayOf(PropTypes.node).isRequired,
+  };
+
+  static defaultProps = {
+    cutoff: 0,
+  };
+
+  constructor(props, context) {
+    super(props, context);
+
+    this.state = {
+      isOpen: false,
+    };
+  }
+
+  handleClick = e => {
+    e.preventDefault();
+    this.setState({isOpen: !this.state.isOpen});
+  };
+
+  render() {
+    const {labelOpen, labelClosed, cutoff} = this.props;
+    const children = React.Children.toArray(this.props.children);
+
+    if (React.Children.count(children) <= cutoff + 1) return children;
```

Comment:
Do you have to use `React.Children.count` if we've already transformed it into a guaranteed array

Reply:
That is a great question. @billyvg suggested both and I just did it like a true solider lol.

[This post](https://stackoverflow.com/questions/34466068/how-to-check-how-many-react-children-have-a-certain-prop) seems to imply there might be some hidden flexibility issues but when I mucked with the fiddle I couldn't get anything different to come out. Maybe @billyvg knows?

Reply:
Yeah I think operating on the array should be fine